src/main/java/com/exasol/adapter/installer/dialect/DialectProperty.java

```diff
@@ -0,0 +1,64 @@
+package com.exasol.adapter.installer.dialect;
+
+import java.util.Objects;
+
+import com.exasol.adapter.installer.UserInputValidator;
+
+/**
+ * Represents a single dialect property.
+ */
+public class DialectProperty {
+    private final String key;
+    private final String value;
+
+    /**
+     * Instantiate a new {@link DialectProperty}.
+     *
+     * @param key   key
+     * @param value value
+     */
+    public DialectProperty(final String key, final String value) {
+        this.key = UserInputValidator.validatePropertyKey(key);
+        this.value = UserInputValidator.validateLiteralString(value);
+    }
+
+    /**
+     * Get a key.
+     *
+     * @return key
+     */
+    public String getKey() {
+        return this.key;
+    }
+
+    /**
+     * Get a value.
+     *
+     * @return value
+     */
+    public String getValue() {
+        return this.value;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
```

Comment:
```suggestion
        if (!(o isinstanceof DialectProperty)) {
            return false;
        }
```

Reply:
getClass version is usually autogenerated, Effective Java Item 8 suggests to use isinstanceof.

Reply:
The methods are generated by my IDE